flocker/node/test/test_loop.py

```diff
@@ -557,6 +611,38 @@ def test_convergence_done_delays_new_iteration(self, logger):
              [(NodeStateCommand, dict(state_changes=(local_state,)))])
         )
 
+    def test_convergence_done_delays_new_iteration_ack(self):
+        """
+        An FSM completing the changes from one convergence iteration doesn't
+        start another iteration, if the control node hasn't acknowledged the
+        last state update.
```

Comment:
I think the really important intent of this test is that a state update isn't sent if the control node hasn't acked the last state update.  The fact that we do this by delaying the entire subsequent iteration is an implementation detail and probably one we want to change (since introducing a round-trip to the control service for every step of convergence is going to slow down the already very slow convergence process - particularly for the multistep convergence operations like dataset creation).  So the docstring should be clear about this intent and perhaps note the expedient implementation decision that was made (bonus would be if the test also actually made an assertion directly tied to that intent rather than being based on the implementation details as well).